src/variables/utils/getWindowVars.ts

```diff
@@ -227,3 +231,83 @@ export const getVariableForZoomRequery = (
       return variable
   }
 }
+
+/*
+ * Prevents the windowPeriod from being null for the time axis
+ *   this fallback is used only for the zoom re-query feature
+ *   if the windowPeriod cannot be found, apply the user-selected domain
+ *     to the original data set and divide by the optimal number of graph points
+ *     to get the estimated windowPeriod
+ */
+export const normalizeWindowPeriodForZoomRequery = (
+  windowPeriod: number | null,
+  timeRange: TimeRange,
+  domain: Array<number>,
+  column: NumericColumnData | string[]
+): number => {
+  if (windowPeriod || !timeRange || !column) {
```

Comment:
what does `!timeRange` and `column` actually mean? The [type for TimeRange](https://github.com/influxdata/ui/blob/master/src/types/queries.ts#L18) makes it seem like `!timeRange` is something that can't happen? And it looks like `NumericColumnData` is going to be an array, so should it instead be `!column.length`?